### DISCUSSION TITLE
cleanup actions?

### DISCUSSION THREAD

**Discussion by abenz1267** [mfussenegger/nvim-jdtls#405] [Q&A] [answered]
I noticed JDTLS has cleanup actions, see https://github.com/eclipse/eclipse.jdt.ls/pull/2350

Does this work with neovim? I tried setting `cleanup = { actionsOnSave = { "addFinalModifier" } }`  but either this setup is wrong or it's just not supported.

What about organizing imports on save?

Regards

**Comment by mfussenegger** [comment, marked as answer]
Both cleanup and `saveActions = { organizeImports = true, },` should work, but I think they require the `willSaveWaitUntil` capability in the client. This is only supported on neovim 0.9-dev:  https://github.com/neovim/neovim/pull/21315

**Comment by abenz1267** [comment]
Ah, interesting! Tried it. works like a charm. guess i'll be on nightly then until something crucial breaks. thanks!

**Comment by adambasis** [comment]
Is there a way to manually trigger these cleanup actions?
